vendor/src/github.com/docker/engine-api/types/types.go

```diff
@@ -274,6 +274,16 @@ type ExecStartCheck struct {
 	Tty bool
 }
 
+// Health stores information about the container's healthcheck results.
+type Health struct {
+	Status         string    // States are: "starting", "healthy", "unhealthy"
+	FailingStreak  uint64    // Number of consecutive failures
+	LastCheckStart time.Time // Zero if never started
```

Comment:
Create subtype `Last`, with these fields:

```go
type HealthcheckResult {
  Start, End time.Time
  ExitCode int
  Output []byte // string may not be right type.
}
```

Then, use a slice for the latest n results:

```
type Health struct {
  Results []HealthCheckResult
}
```

Contents of results can be managed by policy or tweaked for ideal use case. Things like last N failures can be calculated by looking at entries.